### ISSUE TITLE
ListQueues returns deleted queues

### ISSUE THREAD

**Issue by Coder3333** [aws/aws-sdk-net#2127]
### Describe the bug

AmazonSQSClient.ListQueues is including urls that have been deleted.  These are recently deleted queues, but not too recent.  Sometimes it has been a minute or two since it was deleted, and it is still returned.  This is not a race condition, and it is happening a lot, like 50% of the time.

`           
var existingQueuesTask = sqsClient.ListQueuesAsync("some prefix");
`

### Expected Behavior

1. Create some queues with AmazonSQSClient.CreateQueue.
2. Publish messages to the queues, read the messages, and delete the messages.
3. Use AmazonSQSClient.ListQueues to get a list of the queues that you previously created and call AmazonSQSClient.DeleteQueue to delete the queues.
4. Wait a couple of seconds and call AmazonSQSClient.ListQueues again.  This should not return the queues that you already deleted.

### Current Behavior

1. Create some queues with AmazonSQSClient.CreateQueue.
2. Publish messages to the queues, read the messages, and delete the messages.
3. Use AmazonSQSClient.ListQueues to get a list of the queues that you previously created and call AmazonSQSClient.DeleteQueue to delete the queues.
4. Wait a couple of seconds and call AmazonSQSClient.ListQueues again.  There is a 50% chance it will return the queues that you had just deleted.  When you call AmazonSQSClient.DeleteQueue on the urls, it will throw an exception, "The specified queue does not exist for this wsdl version".

### Reproduction Steps

1. Create some queues with AmazonSQSClient.CreateQueue.
2. Publish messages to the queues, read the messages, and delete the messages.
3. Use AmazonSQSClient.ListQueues to get a list of the queues that you previously created and call AmazonSQSClient.DeleteQueue to delete the queues.
4. Wait a couple of seconds and call AmazonSQSClient.ListQueues again.  There is a 50% chance it will return the queues that you had just deleted.  When you call AmazonSQSClient.DeleteQueue on the urls, it will throw an exception, "The specified queue does not exist for this wsdl version".

### Possible Solution

I have no idea why this would be happening.  Does the client or service cache the call to AmazonSQSClient.ListQueues?

### AWS .NET SDK and/or Package version used

AWSSDK.SQS 3.7.2.101

### Targeted .NET Platform

.Net Core 3.1

### Operating System and version

Windows 10

**Comment by ashishdhingra** [comment]
Hi @Coder3333,

Good afternoon.

Thanks for reporting the issue.

Just to mention here the SDK logic for SQS (for most part) is auto-generated from service models and for most part it's delegating the call to the service API operation. As per [Amazon SQS: DeleteQueue](https://docs.aws.amazon.com/AWSSimpleQueueService/latest/APIReference/API_DeleteQueue.html) service API operation, the deletion process takes up to 60 seconds (or may be more). So you might want to wait for sometime (preferably more than be 60 seconds) after you send request to delete the queues (you might also want to wait for successful delete response for each deleted queue). Is you wait time less than 60 seconds?

Thanks,
Ashish

**Comment by Coder3333** [comment]
@ashishdhingra - I am already waiting for the delete response to complete before calling ListQueues.  Are you saying that after a Delete operation completes, it can still take another 60 seconds before the server actually registers the delete?

**Comment by ashishdhingra** [comment]
> @ashishdhingra - I am already waiting for the delete response to complete before calling ListQueues. Are you saying that after a Delete operation completes, it can still take another 60 seconds before the server actually registers the delete?

@Coder3333 Yes (as per [Amazon SQS: DeleteQueue](https://docs.aws.amazon.com/AWSSimpleQueueService/latest/APIReference/API_DeleteQueue.html)). The behavior is not controlled by the SDK.

**Comment by Coder3333** [comment]
@ashishdhingra - if that is the case, then my real issue is with AmazonSQSClient.DeleteQueue.  It doesn't matter so much that it takes 60 seconds to take effect, but when I pass in a url for a queue that is already deleted, it should quietly ignore it, instead of throwing an exception.
